src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -2789,78 +2793,56 @@ export class ResultsViewPageStore {
-                                    this.studyToMolecularProfileDiscreteCna.result!,
-                                    this.studyIdToStudy,
-                                    this.molecularProfileIdToMolecularProfile,
-                                    this.clinicalDataForSamples,
-                                    this.studiesForSamplesWithoutCancerTypeClinicalData,
-                                    this.germlineConsentedSamples,
-                                    this.indexedHotspotData,
-                                    this.indexedVariantAnnotations,
-                                    this.uniqueSampleKeyToTumorType.result!,
-                                    this.generateGenomeNexusHgvsgUrl,
-                                    this.genomeNexusClient,
-                                    this.genomeNexusInternalClient
-                                );
-                                return map;
-                            },
-                            {}
-                        )
-                    );
-                } else {
-                    return Promise.resolve({});
-                }
+    public createMutationMapperStoreForSelectedGene(gene: Gene) {
+        console.log('created a new store');
+
+        const store = new ResultsViewMutationMapperStore(
+            AppConfig.serverConfig,
+            {
+                filterMutationsBySelectedTranscript: true,
             },
-        },
-        {}
-    );
+            gene,
+            this.samples,
+            this.oncoKbCancerGenes,
+            () => this.mutationsByGene.result![gene.hugoGeneSymbol] || [],
+            () => this.mutationCountCache,
+            () => this.genomeNexusCache,
+            () => this.genomeNexusMutationAssessorCache,
+            () => this.genomeNexusMyVariantInfoCache,
+            () => this.discreteCNACache,
+            this.studyToMolecularProfileDiscreteCna.result!,
+            this.studyIdToStudy,
+            this.molecularProfileIdToMolecularProfile,
+            this.clinicalDataForSamples,
+            this.studiesForSamplesWithoutCancerTypeClinicalData,
+            this.germlineConsentedSamples,
+            this.indexedHotspotData,
+            this.indexedVariantAnnotations,
+            this.uniqueSampleKeyToTumorType.result!,
+            this.generateGenomeNexusHgvsgUrl,
+            this.genomeNexusClient,
+            this.genomeNexusInternalClient
+        );
+        this.mutationMapperStoreByGene[gene.hugoGeneSymbol] = store;
+        return store;
+    }
 
     public getMutationMapperStore(
-        hugoGeneSymbol: string
+        gene: Gene
     ): ResultsViewMutationMapperStore | undefined {
-        return this.mutationMapperStores.result[hugoGeneSymbol];
+        if (
+            this.genes.isComplete &&
+            this.oncoKbCancerGenes.isComplete &&
+            this.oncoKbCancerGenes &&
+            this.uniqueSampleKeyToTumorType &&
+            this.mutations &&
```

Comment:
is there a reason we aren't checking for some mutation isComplete?  same for samples